tests/units/sc2/environments/test_sc2_environment.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+from unittest.mock import patch
+
+from urnai.sc2.environments.sc2environment import SC2Env
+
+
+class TestSC2Environment(unittest.TestCase):
+    @patch('pysc2.lib.features.Dimensions')
+    @patch('pysc2.lib.features.AgentInterfaceFormat')
+    @patch('pysc2.env.sc2_env.SC2Env')
+    def test_start_on_init(self, MockSC2Env, MockAgentInterfaceFormat, MockDimensions):
+        SC2Env()
+        MockSC2Env.assert_called_once()
+        MockAgentInterfaceFormat.assert_called_once()
+        MockDimensions.assert_called_once()
+
+    @patch('pysc2.lib.features.Dimensions')
+    @patch('pysc2.lib.features.AgentInterfaceFormat')
+    @patch('pysc2.env.sc2_env.SC2Env')
+    def test_start_after_init(
+        self, MockSC2Env, MockAgentInterfaceFormat, MockDimensions
+    ):
+        # Even after another call of start, the sc2_env of pysc2
+        # should only be instanciated once.
+        env = SC2Env()
+        env.start()
+        MockSC2Env.assert_called_once()
+        MockAgentInterfaceFormat.assert_called_once()
+        MockDimensions.assert_called_once()
+
+    def test_step_method(self):
+        # TODO
+        pass
+
+    def test_reset_method(self):
+        # TODO
+        pass
+
+    @patch('pysc2.env.sc2_env.SC2Env')
+    def test_close_method(self, MockSC2Env):
+        # TODO: Check why this does not work
+        # env = SC2Env()
+        # env.close()
+        # MockSC2Env.close.assert_called_once()
```

Comment:
I tried to test the `close()` method with this commented code, but it didn't work well. My idea was:
- In `env = SC2Env()`, when inside the method `start()`, instanciate the local variable `env_instance` with the mocked class.
- In `env.close()`, call the `close()` method from `env_instance` (which was instanciated with the mocked class).
- Check if the method `close()` from the mocked class was called only once.

But when testing it out, the test fails, asserting that the method `close()` from the mocked class was never called.

Reply:
I think it worked out. Now the code is like this, and the tests run normally:

```python
@patch('pysc2.env.sc2_env.SC2Env')
    def test_close_method(self, MockSC2Env):
        env = SC2Env()
        env.close()
        env.env_instance.close.assert_called_once()
```

Is that correct?

Reply:
I don't think that's right. I think the correct thing would be something like:

```suggestion
    @patch('pysc2.env.sc2_env.SC2Env')
    def test_close_method(self, scenv_mock):
        # GIVEN
        env = SC2Env()
        # WHEN
        env.close()
        # THEN
        scenv_mock.env_instance.close.assert_called_once()
```

Reply:
I recommend using `given-when-then` when writing your tests. All you have to do is separate your test with the 3 comments I've shown above. Here's a link if you want to dig deeper: [Given-When-Then](https://pythontest.com/strategy/given-when-then-2/)

Reply:
> I don't think that's right. I think the correct thing would be something like:

I tried this code and got the same error as before, `close` is never called.